src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -960,15 +965,25 @@ int xcldev::device::runTestCase(const std::string& py,
         return -ENOENT;
     }
 
-    // Program xclbin first.
-    int ret = program(xclbinPath, 0);
-    if (ret != 0) {
-        output += "ERROR: Failed to download xclbin: ";
-        output += xclbin;
-        return -EINVAL;
-    }
+    std::string cmd;
+
+    if (is_u50) {
+        std::string exePath;
+        searchXsaAndDsa(xsaXclbinPath, dsaXclbinPath, exePath, output);
+        exePath += py;
+        cmd = exePath + " -j " + xclbinPath + " " + " -d " + std::to_string(m_idx);        
+    } else {
+        // Program xclbin first.
```

Comment:
Please program xclbin first before run the test case, even for u50 case. This is a good test case for XRT. It will not change anything for the test.